### PR TITLE
LWDEV-9964 Transfers to contract addresses supported

### PR DESCRIPTION
**Pull request by PaterSantyago**

### CODE REVIEW

File: src/Lykke.Service.EthereumApi.Core/Services/IAddressService.cs [hunk 1]
```diff
@@ -1,10 +1,44 @@
...
+using Lykke.Service.EthereumApi.Core.Domain;
+using Lykke.Service.EthereumCommon.Core.Domain;
 
 namespace Lykke.Service.EthereumApi.Core.Services
 {
     public interface IAddressService
     {
+        Task<AddAddressResult> AddAddressToBlacklistAsync(
+            [NotNull] string address,
+            [NotNull] string reason);
```
Comment (OttoVT) [inline src/Lykke.Service.EthereumApi.Core/Services/IAddressService.cs:14]: May be it is better to store not a string but the code of a reason?

File: src/Lykke.Service.EthereumApi.Services/AddressService.cs [hunk 2]
```diff
@@ -1,28 +1,157 @@
...
+        public async Task<string> TryGetBlacklistingReason(
+            string address)
+        {
+            var blacklistedAddress = await _blacklistedAddressRepository.TryGetAsync(address);
+
+            if (blacklistedAddress != null)
+            {
+                var reason = blacklistedAddress.BlacklistingReason;
+
+                return !string.IsNullOrEmpty(reason) ? reason : "Unspecified";
```
Comment (OttoVT) [inline src/Lykke.Service.EthereumApi.Services/AddressService.cs:124]: 100% sure that there is no "reason" to store string in table. They all will be the same, mostly. (Use Enum for reasons)

Comment (PaterSantyago) [reply src/Lykke.Service.EthereumApi.Services/AddressService.cs:124]: Nope, in some cases token may be blacklisted manually for some custom reason (i.e. for bad reputation)

File: src/Lykke.Service.EthereumApi.Core/Services/IAddressService.cs (hunk 1 above)
Comment (PaterSantyago) [reply src/Lykke.Service.EthereumApi.Core/Services/IAddressService.cs:14]: Nope, in some cases token may be blacklisted manually for some custom reason (i.e. for bad reputation). Moreover, we store in reason hashes of failed transactions, estimated and max gas amounts and so on.

**Review by OttoVT — APPROVED** [review]

**Pull request merged**